### PR TITLE
DSL2 Modules - Developer Tutorial

### PR DESCRIPTION
**Pull request by lescai**
This resolves issue #713 
I've read through, and hopefully there aren't additional typos but would be helpful if someone could skim through briefly 😇

### CODE REVIEW

**Review by ewels — COMMENTED** [review]
Love it! A few minor pedantic suggestions but I love the way it reads, super nice! 🚀 

The screenshots are beautiful, but part of me wonders if it's better to do code blocks using `console` formatting? eg:

```console
$ nf-core modules do-something

INFO   I am some example output
INFO   It worked!!
```

That's what I usually do in these kinds of documents, but is a bit of a faff (I usually edit off the line numbers that Rich adds and of course you lose the nice colours). So only opening it for discussion rather than suggesting you change anything here.

**Comment by ewels** [comment]
## Markdown linting is failing

To keep the code consistent with lots of contributors, we run automated code consistency checks.
To fix this CI test, please run:

* Install `markdownlint-cli`
    * On Mac: `brew install markdownlint-cli`
    * Everything else: [Install `npm`](https://www.npmjs.com/get-npm) then [install `markdownlint-cli`](https://www.npmjs.com/package/markdownlint-cli) (`npm install -g markdownlint-cli`)
* Fix the markdown errors
    * Automatically: `markdownlint . --config .github/markdownlint.yml --fix`
    * Manually resolve anything left from `markdownlint . --config .github/markdownlint.yml`

Once you push these changes the test should pass, and you can hide this comment :+1:

We highly recommend setting up markdownlint in your code editor so that this formatting is done automatically on save. Ask about it on Slack for help!

Thanks again for your contribution!

File: markdown/tutorials/dsl2_modules_tutorial.md [hunk 1]
```diff
@@ -0,0 +1,438 @@
...
+```
+The branch will be syncronised with your remote once you push the first new commit.
+
+- you can use the github interface
+
+To do this, you can select the dropdown menu on the top-left of your repository code, write the name of the new branch and choose to create it as shown in the figure below.
+
+![branch](assets/dsl2-mod_02_new_branch.png)
+
+You will then sync this locally (ideally, you clone the forked repository on your working environment to edit code more comfortably)
```
[the remark this answers is not in the document]
Comment (lescai) [reply markdown/tutorials/dsl2_modules_tutorial.md]: I've accepted the suggestions and committed it from local, to apply it on the updated file with lint corrections.

File: markdown/tutorials/dsl2_modules_tutorial.md [hunk 2]
```diff
@@ -0,0 +1,438 @@
...
+
+---
+## Create the module template
+
+Using *nf-core tools* it is very easy to create a new module. In our example, we change directory into the repository (*modules*) and we type
+
+```
+nf-core modules create -t fgbio/fastqtobam
+```
+You will notice that with `-t` we indicate the name of the new module we want to create using the following syntax:
```
[the remark this answers is not in the document]
Comment (lescai) [reply markdown/tutorials/dsl2_modules_tutorial.md]: fixed now, it lints with markdownlint-cli

**Review by ewels — APPROVED** [review]
Great stuff! Next time you can add the suggestions to a batch and apply them in one commit, just so you know 😉

**Comment by ewels** [comment]
I had to tweak a bunch of stuff to make this actually work on the website, for future reference: https://github.com/nf-core/nf-co.re/commit/e0f68beecce8f0dc3524bc68aea6679cdf84086b

* Move the images to a directory where they can be viewed on the website and update the paths
* Add the required YAML front-matter for the page
* Add link and description to the tutorial from the tutorials index page

**Comment by ewels** [comment]
Tutorial now live at: https://nf-co.re/developers/tutorials/dsl2_modules_tutorial

**Pull request merged**